# Who Is J?

## Analysing JOTB diversity network 

One of the main goals of the ‘Yes We Tech’ community is contributing to create an inclusive space where we can celebrate diversity, provide visibility to women-in-tech, and ensure that everybody has an equal chance to learn, share and enjoy technology-related disciplines.

As co-organisers of the event, we have concentrated our efforts in getting more women speakers on board under the assumption that a more diverse panel would enrich the conversation also around technology.

Certainly, we have doubled the number of women giving talks this year, but, is this diversity enough? How can we know that we have succeeded in our goal? and more importantly, what can we learn to create a more diverse event in future editions?

The work that we are sharing here talks about two things: data and people. Both data and people should help us to find out some answers and understand the reasons why.

Let's start with a story about data. Data is pretty simple compared with people. Just take a look at the numbers, the small ones, the ones that better describe what happened in 2016 and 2017 J On The Beach editions.

In [185]:
import pandas as pd
import numpy as np
import scipy as sp
import pygal
from iplotter import GCPlotter

plotter = GCPlotter()

### Small data analysis

Small data says that last year, our 'J' engaged up to 48 speakers and 299 attendees into this big data thing. 
I'm not considering here any member of the organisation.

In [210]:
data2016 = pd.read_csv('../input/small_data_2016.csv')
data2016['Women Rate'] = pd.Series(data2016['Women']*100/data2016['Total'])
data2016['Men Rate'] = pd.Series(data2016['Men']*100/data2016['Total'])
data2016

Tribe  Women  Men  Total  Women Rate    Men Rate
0                speakers      5   43     48   10.416667   89.583333
1               attendees     39  260    299   13.043478   86.956522
2             independent      8   44     52   15.384615   84.615385
3           company_teams     28  214    242   11.570248   88.429752
4  company_teams_no_women      0   99     99    0.000000  100.000000
5               hackathon      0    0      0         NaN         NaN

This year speakers are 40, few less than last year, while participation have reached the number of 368 people. (Compare the increment of attendees 368 vs 299

In [211]:
data2017 = pd.read_csv('../input/small_data_2017.csv')
data2017['Women Rate'] = pd.Series(data2017['Women']*100/data2017['Total'])
data2017['Men Rate'] = pd.Series(data2017['Men']*100/data2017['Total'])
data2017

Tribe  Women  Men  Total  Women Rate    Men Rate
0                speakers     11   29     40   27.500000   72.500000
1               attendees     36  332    368    9.782609   90.217391
2             independent      6   65     71    8.450704   91.549296
3           copmany_teams     30  267    297   10.101010   89.898990
4  company_teams_no_women      0  134    134    0.000000  100.000000
5               hackathon      4   21     25   16.000000   84.000000

In [8]:
increase = 100 - 299*100.00/368
increase

18.75

It is noticable also, that big data is bigger than ever and this year we have included workshops and a hackathon.
  
The more the better right? Let's continue because there are more numbers behind those ones. Numbers that will give us some signs of diversity.

#### Diversity

When it comes about speakers, this year we have a **27.5%** of women speaking to J, compared with a rough **10.4%** of the last year.

In [212]:
data = [
    ['Tribe', 'Women', 'Men', {"role": 'annotation'}],
    ['2016', data2016['Women Rate'][0], data2016['Men Rate'][0],''],
    ['2017', data2017['Women Rate'][0], data2017['Men Rate'][0],''],
]
options = {
    "title": 'Speakers at JOTB',
    "width": 600,
    "height": 400,
    "legend": {"position": 'top', "maxLines": 3},
    "bar": {"groupWidth": '50%'},
    "isStacked": "true",
    "colors": ['#984e9e', '#ed1c40'],
}

plotter.plot(data,chart_type='ColumnChart',chart_package='corechart', options=options)

However, and this is the worrying thing, the participation of women as attendees has slightly dropped from a not too ambitious **13%** to a disappointing **9.8%**. So we have an x% more of attendees but zero impact on a wider variaty of people.

In [213]:
data = [
    ['Tribe', 'Women', 'Men', {"role": 'annotation'}],
    ['2016', data2016['Women Rate'][1], data2016['Men Rate'][1],''],
    ['2017', data2017['Women Rate'][1], data2017['Men Rate'][1],''],
]
options = {
    "title": 'Attendees at JOTB',
    "width": 600,
    "height": 400,
    "legend": {"position": 'top', "maxLines": 3},
    "bar": {"groupWidth": '55%'},
    "isStacked": "true",
    "colors": ['#984e9e', '#ed1c40'],
}

plotter.plot(data,chart_type='ColumnChart',chart_package='corechart', options=options)

#### Why this happened? 

We don’t really know. But we continued looking at the numbers and realised that **30** of the **45** companies that enrolled two or more people didn't include any women on their lists. Meaning a **31%** of the mass of attendees. Correlate team size with women percentage to validate if: the smaller the teams are, the less chances to include a women on their lists

In [214]:
companies_team = data2017['Total'][3] + data2017['Total'][4]
mass_represented = pd.Series(data2017['Total'][4]*100/companies_team)
women_represented = pd.Series(100 - mass_represented)
mass_represented

0    31
dtype: int64

For us this is not a good sign. Despite the fact that our ability to summon has increased on our monthly meetups (the ones that attempts to create this culture for equality on Málaga), the engagement on other events doesn’t have a big impact.

Again I'm not blaming companies here, because if we try to identify the participation rate of women who are not part of a team, the representation also decreased almost a **50%**.

In [215]:
data = [
    ['Tribe', 'Women', 'Men', {"role": 'annotation'}],
    [data2016['Tribe'][2], data2016['Women Rate'][2], data2016['Men Rate'][2],''],
    [data2016['Tribe'][3], data2016['Women Rate'][3], data2016['Men Rate'][3],''],
    [data2016['Tribe'][5], data2016['Women Rate'][5], data2016['Men Rate'][5],''],
]
options = {
    "title": '2016 JOTB Edition',
    "width": 600,
    "height": 400,
    "legend": {"position": 'top', "maxLines": 3},
    "bar": {"groupWidth": '55%'},
    "isStacked": "true",
    "colors": ['#984e9e', '#ed1c40'],
}

plotter.plot(data,chart_type='ColumnChart',chart_package='corechart', options=options)

In [23]:
data = [
    ['Tribe', 'Women', 'Men', {"role": 'annotation'}],
    [data2017['Tribe'][2], data2017['Women Rate'][2], data2017['Men Rate'][2],''],
    [data2017['Tribe'][3], data2017['Women Rate'][3], data2017['Men Rate'][3],''],
    [data2017['Tribe'][5], data2017['Women Rate'][5], data2017['Men Rate'][5],''],
]
options = {
    "title": '2017 JOTB Edition',
    "width": 600,
    "height": 400,
    "legend": {"position": 'top', "maxLines": 3},
    "bar": {"groupWidth": '55%'},
    "isStacked": "true",
    "colors": ['#984e9e', '#ed1c40'],
}

plotter.plot(data,chart_type='ColumnChart',chart_package='corechart', options=options)

Before before blaming anyone or falling to quickly into self-indulgence, there are still more data to play with.

Note aside: the next thing is nothing but an experiment, nothing is categorical or has been made with the intention of offending any body. Like our t-shirt labels says: no programmer have been injured in the creation of the following data game.

# Social network analysis
The next story talks about people. The people around J, the ones who follow, are followed by, interact with, and create the chances of a more diverse and interesting conference. 

It is also a story about the people who organise this conference. Because when we started to plan a conference like this, we did nothing but thinking on what could be interesting for the people who come. In order to get that we used the previous knowledge that we have about cool people who do amazing things with data, and JVM technologies. And this means looking into our own networks and following suggestions of the people we trust. 

So if we assume that we are biased by the people around us, we thought it was a good idea to know first how is the network of people around J to see the chances that we have to bring someone different, unusual that can add value to the conference.

For the moment, since this is an experiment that wants to trigger your reaction we will look at J's Twitter account.

Indeed, a real-world network would have a larger amount of numbers and people to look at, but yet a digital social network is about human interactions, conversations and knowledge sharing. 

For this experiment we've used `sexmachine` python library https://pypi.python.org/pypi/SexMachine/ and the 'Twitter Gender Distribution' project published in github https://github.com/ajdavis/twitter-gender-distribution to find out the gender of a specific twitter acount.

In [319]:
run index.py jotb17

C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\udca1' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\udc7d' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\udf4e' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\udf47' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\udf53' will be ignored. You might be using a narrow Python build.
  return _unidecod

From the small **50%** of J's friends that could be identified with a gender, the distribution woman/men is a **20/80**. Friends are the ones who follow and are followed by J.

In [158]:
whoisj = pd.read_json('../out/jotb17.json', orient = 'columns')
whoisj['jotb17']['female_rate']
whoisj['jotb17']['male_rate']
whoisj['jotb17']['nonbinary_rate']
whoisj

jotb17
favourites_count                                               1272
female_count                                                    175
female_rate                                                     14%
followers_count                                                1138
followers_list    {u'Jodoniwi': {u'lang': u'es', u'favourites_co...
friends_count                                                   177
friends_list      {u'rgransberger': {u'lang': u'de', u'favourite...
gender                                                 undetermined
id                                                       3899375963
lang                                                             es
location                                             Málaga, España
male_count                                                      546
male_rate                                                       44%
name                                                 J On The Beach
nonbinary_count                                                   5
nonbinary_rate                                                   0%
statuses_count                                                 1228
total_count                                                    1233
undefined_count                                                 507
undefined_rate                                                  41%

In [159]:
friends_total = whoisj['jotb17']['friends_count']
friends_total

177

In [160]:
followers_total = whoisj['jotb17']['followers_count']
followers_total

1138

In [161]:
people = pd.read_json(whoisj['jotb17'].to_json())
people

favourites_count  female_count female_rate  followers_count  \
1968damasco                  1272           175         14%             1138   
47deg                        1272           175         14%             1138   
4dokoneko                    1272           175         14%             1138   
4lberto                      1272           175         14%             1138   
666Anastazja999              1272           175         14%             1138   
6P592Otv                     1272           175         14%             1138   
ADRYVERA27                   1272           175         14%             1138   
AIOM_oficial                 1272           175         14%             1138   
APA42                        1272           175         14%             1138   
APC_AssoFrance               1272           175         14%             1138   
AUTOEMPLEOEXITO              1272           175         14%             1138   
AbadArmijos                  1272           175         14%             1138   
AbiPrincStar                 1272           175         14%             1138   
AcademiasUnitec              1272           175         14%             1138   
Addknowledge                 1272           175         14%             1138   
AdeSorteos                   1272           175         14%             1138   
Adele_Gokkaya                1272           175         14%             1138   
Adelma85                     1272           175         14%             1138   
Adler_Sinclair               1272           175         14%             1138   
AfonsoCorreia6               1272           175         14%             1138   
AgileAlex                    1272           175         14%             1138   
AgileNightmare               1272           175         14%             1138   
Alberete1                    1272           175         14%             1138   
AlbertoSainz2                1272           175         14%             1138   
Alecrin68                    1272           175         14%             1138   
AlejandroG85                 1272           175         14%             1138   
Alejandrofl                  1272           175         14%             1138   
Aleksan10386720              1272           175         14%             1138   
AlesiaBlond                  1272           175         14%             1138   
Alisa_I80                    1272           175         14%             1138   
...                           ...           ...         ...              ...   
wielkiowiec                  1272           175         14%             1138   
willhillpeople               1272           175         14%             1138   
wkdliew                      1272           175         14%             1138   
wojdyga                      1272           175         14%             1138   
wolf4ood                     1272           175         14%             1138   
womenintech                  1272           175         14%             1138   
wonderwomentech              1272           175         14%             1138   
wu_ming_80                   1272           175         14%             1138   
xDomi10                      1272           175         14%             1138   
xantic9                      1272           175         14%             1138   
xcoderteam                   1272           175         14%             1138   
xeraa                        1272           175         14%             1138   
xiconormal                   1272           175         14%             1138   
xzakariye16                  1272           175         14%             1138   
yanaga                       1272           175         14%             1138   
yarasenger                   1272           175         14%             1138   
yeswetech_                   1272           175         14%             1138   
yol                          1272           175         14%             1138   
yxf07            

In [162]:
followers = pd.read_json(people['followers_list'].to_json(), orient = 'index')
followers

favourites_count  followers_count  friends_count  \
1968damasco                   1.0             19.0          302.0   
47deg                      2122.0           3208.0          466.0   
4dokoneko                   268.0            669.0          238.0   
4lberto                    2420.0            435.0          485.0   
666Anastazja999             574.0            254.0         1098.0   
6P592Otv                   4338.0           1015.0         4730.0   
ADRYVERA27                   11.0            578.0         2741.0   
AIOM_oficial                 77.0            586.0         2754.0   
APA42                       964.0            401.0          699.0   
APC_AssoFrance              450.0           1083.0         4373.0   
AUTOEMPLEOEXITO             314.0           1605.0         5001.0   
AbadArmijos                3562.0           1022.0         3982.0   
AbiPrincStar                 77.0             26.0          201.0   
AcademiasUnitec             225.0            367.0         1422.0   
Addknowledge                 56.0           5808.0         5720.0   
AdeSorteos                   16.0            316.0         2471.0   
Adele_Gokkaya               170.0            290.0         1144.0   
Adelma85                    218.0            165.0          355.0   
Adler_Sinclair            25940.0           5614.0         5193.0   
AfonsoCorreia6               21.0             23.0          100.0   
AgileAlex                   182.0             57.0          246.0   
AgileNightmare               30.0           1609.0         3346.0   
Alberete1                     0.0            730.0         4796.0   
AlbertoSainz2                27.0            265.0          862.0   
Alecrin68                 31785.0           1397.0         1415.0   
AlejandroG85                 27.0             94.0          228.0   
Alejandrofl                1393.0            246.0          762.0   
Aleksan10386720               9.0           1179.0         4543.0   
AlesiaBlond                   7.0            269.0          701.0   
Alisa_I80                   504.0            233.0         1150.0   
...                           ...              ...            ...   
wielkiowiec                   0.0              5.0           32.0   
willhillpeople              205.0           1290.0          462.0   
wkdliew                     214.0             10.0         1711.0   
wojdyga                    1296.0            141.0          979.0   
wolf4ood                    294.0            163.0          591.0   
womenintech                   NaN              NaN            NaN   
wonderwomentech               NaN              NaN            NaN   
wu_ming_80                 8787.0           1802.0         4164.0   
xDomi10                    2095.0             57.0          149.0   
xantic9                   39726.0            142.0         1896.0   
xcoderteam                  495.0             65.0          447.0   
xeraa                      2278.0           1235.0          469.0   
xiconormal                   23.0             59.0          523.0   
xzakariye16                   2.0             96.0          586.0   
yanaga                        NaN              NaN            NaN   
yarasenger                 4003.0           6066.0          846.0   
yeswetech_                  514.0            632.0          507.0   
yol                         304.0            808.0          498.0   
yxf07                       128.0              4.0          128.0   
zdenka_markotic            3076.0            192.0          632.0   
zeitiger                    770.0            122.0          278.0   
zisk0                       731.0            296.0          451.0   
zjmruiz666                  572.0             56.0          768.0   
zlatakristina5            64589.0           1495.0         4955.0   
zoltan_bence                  0.0              1.0            8.0   
zonadevalor                 255.0             91.0          216.0   
zosimmargaritov

In [216]:
followers['gender'].value_counts()

male             474
undetermined     454
female           135
mostly_female     31
mostly_male       29
nonbinary          4
Name: gender, dtype: int64

In [296]:
followers_dist = followers['gender'].value_counts()

followers_map = pygal.Treemap(height=400)
followers_map.title = 'Followers Gender Map'
followers_map.add('Male',followers_dist[0])
followers_map.add('Undetermined',followers_dist[1])
followers_map.add('Female',followers_dist[2])
followers_map.add('Mostly Female',followers_dist[3])
followers_map.add('Mostly Male',followers_dist[4])
followers_map.add('Non binary',followers_dist[5])

followers_map.render_in_browser()

file://c:/users/carme/appdata/local/temp/tmpltyjzc.html


In [252]:
lang_counts = followers['lang'].value_counts()
languages = followers['lang'].value_counts().keys()

followers_dist = followers['gender'].value_counts()

lang_followers_map = pygal.Treemap(height=400)
lang_followers_map.title = 'Followers Language Map'

for i in languages:
    lang_followers_map.add(i,lang_counts[i])

lang_followers_map.render_in_browser()

file://c:/users/carme/appdata/local/temp/tmpx6frvp.html


In [115]:
followers['location'].value_counts()

                                  303
Málaga                             39
Málaga, España                     21
Madrid                             20
Spain                              13
España                             11
Málaga, Andalucía                  10
London, England                     9
Málaga, Spain                       9
London                              9
Malaga                              8
Portugal                            8
Украина                             7
Madrid, Spain                       7
Madrid, Comunidad de Madrid         6
Malaga, Spain                       6
Sevilla                             5
Porto, Portugal                     5
Almería, España                     4
Seattle, WA                         4
San Francisco                       4
New York, USA                       4
Bulgaria                            4
Dublin City, Ireland                4
Barcelona                           4
Stockholm, Sweden                   3
San Francisc

In [107]:
following = pd.read_json(people['friends_list'].to_json(), orient = 'index')
following

favourites_count  followers_count  friends_count  \
1968damasco                   NaN              NaN            NaN   
47deg                      2122.0           3208.0          466.0   
4dokoneko                     NaN              NaN            NaN   
4lberto                       NaN              NaN            NaN   
666Anastazja999               NaN              NaN            NaN   
6P592Otv                      NaN              NaN            NaN   
ADRYVERA27                    NaN              NaN            NaN   
AIOM_oficial                  NaN              NaN            NaN   
APA42                         NaN              NaN            NaN   
APC_AssoFrance                NaN              NaN            NaN   
AUTOEMPLEOEXITO               NaN              NaN            NaN   
AbadArmijos                   NaN              NaN            NaN   
AbiPrincStar                  NaN              NaN            NaN   
AcademiasUnitec               NaN              NaN            NaN   
Addknowledge                  NaN              NaN            NaN   
AdeSorteos                    NaN              NaN            NaN   
Adele_Gokkaya                 NaN              NaN            NaN   
Adelma85                      NaN              NaN            NaN   
Adler_Sinclair                NaN              NaN            NaN   
AfonsoCorreia6                NaN              NaN            NaN   
AgileAlex                     NaN              NaN            NaN   
AgileNightmare                NaN              NaN            NaN   
Alberete1                     NaN              NaN            NaN   
AlbertoSainz2                 NaN              NaN            NaN   
Alecrin68                     NaN              NaN            NaN   
AlejandroG85                  NaN              NaN            NaN   
Alejandrofl                   NaN              NaN            NaN   
Aleksan10386720               NaN              NaN            NaN   
AlesiaBlond                   NaN              NaN            NaN   
Alisa_I80                     NaN              NaN            NaN   
...                           ...              ...            ...   
wielkiowiec                   NaN              NaN            NaN   
willhillpeople              205.0           1290.0          462.0   
wkdliew                       NaN              NaN            NaN   
wojdyga                       NaN              NaN            NaN   
wolf4ood                      NaN              NaN            NaN   
womenintech                 576.0         111043.0        23210.0   
wonderwomentech            5460.0           5703.0         2716.0   
wu_ming_80                    NaN              NaN            NaN   
xDomi10                       NaN              NaN            NaN   
xantic9                       NaN              NaN            NaN   
xcoderteam                    NaN              NaN            NaN   
xeraa                      2278.0           1235.0          469.0   
xiconormal                    NaN              NaN            NaN   
xzakariye16                   NaN              NaN            NaN   
yanaga                      969.0           4312.0          545.0   
yarasenger                 4003.0           6066.0          846.0   
yeswetech_                  514.0            632.0          507.0   
yol                           NaN              NaN            NaN   
yxf07                         NaN              NaN            NaN   
zdenka_markotic               NaN              NaN            NaN   
zeitiger                      NaN              NaN            NaN   
zisk0                         NaN              NaN            NaN   
zjmruiz666                    NaN              NaN            NaN   
zlatakristina5                NaN              NaN            NaN   
zoltan_bence                  NaN              NaN            NaN   
zonadevalor                   NaN              NaN            NaN   
zosimmargaritov

In [108]:
following['gender'].value_counts()

undetermined     86
male             68
female           13
mostly_female     5
mostly_male       4
nonbinary         1
Name: gender, dtype: int64

In [112]:
following['lang'].value_counts()

en       120
es        37
fr         8
de         5
pt         2
it         2
hu         1
ru         1
en-gb      1
Name: lang, dtype: int64

In [113]:
following['location'].value_counts()

                                        32
London                                   9
San Francisco, CA                        7
Madrid                                   5
Málaga                                   4
Málaga, Spain                            3
Worldwide                                3
Madrid, Comunidad de Madrid              3
Global                                   3
Seattle, WA                              2
Madrid, Spain                            2
London, England                          2
France                                   2
Málaga, Andalucía                        2
Málaga, España                           2
Cambridge, England                       2
London, UK                               2
Germany                                  2
Switzerland                              2
Maringá, Brasil                          1
Ottignies-Louvain-la-Neuve, Belgique     1
Genève, Suisse                           1
In JUGs around the world!                1
Baños del C

### Tweets analysis

In [218]:
run tweets.py yeswetech_ 1200

C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\ud83c' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\uddeb' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\uddf7' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\uddea' will be ignored. You might be using a narrow Python build.
  return _unidecode(string)
C:\Users\carme\Anaconda2\lib\site-packages\unidecode\__init__.py:50: RuntimeWarning: Surrogate character u'\uddfa' will be ignored. You might be using a narrow Python build.
  return _unidecod

In [219]:
j_network = pd.read_json('../out/yeswetech__tweets.json', orient = 'index')
j_network

description  \
Adolfo_fdz       Business Development Manager Spain & Portugal ...   
Andreitakas      Hablo por los codos. Docente y consultora de #...   
AnjanaVakil      Lover of languages, for humans and machines. P...   
AnneliseGripp    Especialista Transformação Ágil,Consultora em ...   
AzaharVyb                                                      ...   
Beatrizhll       Love music, dancing & traveling. UX Research &...   
BicEuronova      Apoyamos la creación, incubación y consolidaci...   
CCComUMA         Cuenta oficial de la Facultad de Ciencias de l...   
CrisRojoLu       All my Tweets are my own view. Love TECH, lear...   
DiarioSUR        #Noticias de #últimahora de #Málaga, España y ...   
DoFideas         @valo_io team | T-cracia https://t.co/dKZ5e4qR...   
EburyES          Especialistas en transacciones de #divisas. No...   
ElHuffPost       ▶Facebook https://t.co/jykEKc6lz9            \...   
ElMundoZen       Suplemento de #bienestar de @elmundoes #wellne...   
EmprendeMadres   Asociación Nacional de Madres Emprendedoras en...   
EncuentrosTech   Agenda, formación y eventos 2.0 de Málaga y pr...   
EscarlataGlez    Española + Finlandesa. #Formacion #TIC #RRHH #...   
GDGMalaga        Google Developer Group Málaga. Conferencias, e...   
GemaiSanchez     Apostando por la innovación, el turismo y la e...   
GirlsinTechES    Queremos hacer más visibles a las #mujeres en ...   
IgualdadURJC     Observatorio de Igualdad de Género de la Unive...   
IgualdadUma      Recién nacidos y creciendo. Grupo pro igualdad...   
Igualdad_UMA                                                         
InfoUMA          Campus Excelencia Internacional @AndaluciaTech...   
InformaticaUMA   Twitter oficial de la E.T.S. Ingeniería Inform...   
JOTB17           An international rendezvous that brings develo...   
LeighaNotLeia    Developer @hubba. I spend my life on the compu...   
LeticiaDolera    Actriz, directora, feminista y mata-zombies pr...   
LizDobsonUoH     Senior Lecturer in Music The Uni of Huddersfie...   
LopSa_Lau        Nací en Málaga y algunos dicen que seseo. Cons...   
...                                                            ...   
mcorreafdez                                                          
mikilla_cs       El amor es eterno o no es. Programadora, escri...   
minichaplin      Programas educativos, proyectos de gestión cul...   
mjcachon         SEO y Marketing Online, friki máxima de Michae...   
nachogil         Professional designer and amateur underwater p...   
opiniondemalaga  Cuenta oficial del diario La Opinión de Málaga...   
pablitomusa      Educational Engineer @elastic #BigData, #Analy...   
pamplonaswcraft             Pamplona Software Craftsmanship #scpna   
petezin          Corresponsal de @el_pais en Silicon Valley / S...   
randommood       Engineering @fastly. Plagued by many interests...   
rtve             Canal oficial en Twitter de RTVE, Radio Televi...   
rvagg                                                                
skamille         Camille of the House Fournier, First of Her Na...   
sorprendida      Cofounder & CEO @elartedemedir\nSocia @kschool...   
soyDelia         Bloguera en Madre 3.0, creadora de @MadresEnre...   
stemxion         Creatividad tecnológica. Pre-ingeniería y #Rob...   
stfn42           Head of Offensive Information Security at @sci...   
tanaligato       SEO, manoseadora de código, perfume junkie. Co...   
tatianeaguirres  Tatiane Aguirres Nogueira: Gaúcha, Estudante d...   
tchellita        Professora, blogueira, facilitadora e mediador...   
theeremita       Acoustic and audio engineer (de Teleco y de to...   
trisha_gee       Coder/blogger/speaker, working for JetBrains. ...   
txapao           #ASO & #SEO Global Head of Organic Acquisition...   
ubuntu                         Official Twitter Account for Ubuntu   
valo_io          Valo is a #streaming #bigdata #analytics platf...   
womenwhodesign   A Twitter profile directory for inspiring wome..

In [220]:
j_network['gender'].value_counts()

undetermined     76
female           34
male             12
mostly_female     7
nonbinary         3
mostly_male       1
Name: gender, dtype: int64

In [317]:
import operator
pairs = []
for i in j_network['gender'].keys() :
    if (j_network['hashtags'][i] != []) : 
        pairs.append([j_network['hashtags'][i], j_network['gender'][i]]) 

key_pairs = []
for i,j in pairs:
    for x in i:
        key_pairs.append((x,j))

key_pairs
key_pair_dist = {x: key_pairs.count(x) for x in key_pairs}
sorted_x = sorted(key_pair_dist.items(), key = operator.itemgetter(1), reverse = True)
sorted_x

[((u'BigData', u'undetermined'), 10),
 ((u'1Abril', u'undetermined'), 10),
 ((u'IoT', u'undetermined'), 7),
 ((u'JOTB17', u'undetermined'), 6),
 ((u'Taller', u'undetermined'), 4),
 ((u'bigdata', u'undetermined'), 4),
 ((u'WomenTechmakers', u'undetermined'), 4),
 ((u'1Abril', u'female'), 4),
 ((u'SEO', u'undetermined'), 4),
 ((u'distributedsystems', u'undetermined'), 3),
 ((u'wtmmalaga2017', u'undetermined'), 3),
 ((u'python', u'undetermined'), 3),
 ((u'data', u'undetermined'), 2),
 ((u'frasesdemadres', u'undetermined'), 2),
 ((u'mujeres', u'undetermined'), 2),
 ((u'womenintech', u'undetermined'), 2),
 ((u'tecnolog\xeda', u'female'), 2),
 ((u'gendergap', u'undetermined'), 2),
 ((u'YWT', u'undetermined'), 2),
 ((u'YesWeTech', u'undetermined'), 2),
 ((u'frasesdemadres', u'female'), 2),
 ((u'Taller', u'female'), 2),
 ((u'uxspain', u'undetermined'), 2),
 ((u'JOTB17', u'male'), 2),
 ((u'JOTB17', u'female'), 1),
 ((u'scala', u'nonbinary'), 1),
 ((u'InteligenciaArtifical', u'undetermined'), 1)

In [293]:
a = j_network['hashtags']
b = j_network['gender']

say_something = [x for x in a if x != []]

tags = []

for y in say_something:
    for x in pd.DataFrame(y)[0]:
        tags.append(x.lower())
        
        
pd.DataFrame(tags)[0].value_counts()

1abril                       14
bigdata                      14
jotb17                        9
iot                           7
taller                        6
seo                           5
womentechmakers               5
tecnología                    4
frasesdemadres                4
yeswetech                     3
wtmmalaga2017                 3
python                        3
distributedsystems            3
womenintech                   3
uxspain                       2
gendergap                     2
data                          2
sonido3d                      2
ywt                           2
opensouthcode                 2
mujeres                       2
microservices                 1
cav                           1
css                           1
charlas                       1
programación                  1
mujeryciencia                 1
yesyoutech                    1
scala                         1
yesshecan                     1
                             ..
desayuno

## Credits

Few lines to credit this work. Thanks M. Carmen Correa to find the time between work and family to collect all these data, coding it in Python and dealing with the Twitter API. Thanks also to Ángela Dini and Gema Sánchez, to keep this project energised and share it with the press and the community. Thanks also to the women who have joined not just once, or twice but many times to Yes We Tech meetups, and for sure thank you for your interest, your support and your time. If I have one credit is just the attempt to organise a space free of the same old-boring-macho thing. Hope you enjoyed it and thank you.

Shared in github https://github.com/YesWeTech/whoIsJ